jar.go

```diff
@@ -292,10 +292,22 @@ func (j *Jar) cookies(u *url.URL, now time.Time) (cookies []*http.Cookie) {
 	}
 
 	sort.Sort(byPathLength(selected))
-	for _, e := range selected {
-		cookies = append(cookies, &http.Cookie{Name: e.Name, Value: e.Value})
-	}
+	return toCookies(selected)
+}
 
+func toCookies(entries []entry) []*http.Cookie {
+	cookies := make([]*(http.Cookie), len(entries))
+	for i, e := range entries {
+		cookies[i] = &http.Cookie{
+			Name:     e.Name,
+			Value:    e.Value,
+			Path:     e.Path,
```

Comment:
Interesting. Looks like we were dropping everything other than `Name` and `Value` before. And, it looks like net/http does the same here: https://golang.org/src/net/http/cookie.go?s=439:952#L219

What is the reason for preserving them?